### PR TITLE
docs: add changelog and updating entries for 1.5.0

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Add `CHANGELOG.md` entries that were released fo 1.5.0 and move the UPDATING entries to the appropriate version.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: UPDATING.md [hunk 1]
```diff
@@ -46,6 +46,15 @@ assists people when migrating to a new version.
 - [18976](https://github.com/apache/superset/pull/18976): When running the app in debug mode, the app will default to use `SimpleCache` for `FILTER_STATE_CACHE_CONFIG` and `EXPLORE_FORM_DATA_CACHE_CONFIG`. When running in non-debug mode, a cache backend will need to be defined, otherwise the application will fail to start. For installations using Redis or other caching backends, it is recommended to use the same backend for both cache configs.
```
Comment (dpgaspar) [inline UPDATING.md:46]: Found that this one is included on 1.5 branch, can you check?

File: UPDATING.md [hunk 2]
```diff
@@ -46,6 +46,15 @@ assists people when migrating to a new version.
...
 - [18970](https://github.com/apache/superset/pull/18970): The `DISABLE_LEGACY_DATASOURCE_EDITOR` feature flag is now `True` by default which disables the legacy datasource editor from being shown in the client.
 - [17984](https://github.com/apache/superset/pull/17984): The default Flask SECRET_KEY has changed for security reasons. You should always override with your own secret. Set `PREVIOUS_SECRET_KEY`, e.g. `PREVIOUS_SECRET_KEY = "\2\1thisismyscretkey\1\2\\e\\y\\y\\h"`, with your previous key and use `superset re-encrypt-secrets` to rotate you current secrets.
```
Comment (dpgaspar) [inline UPDATING.md:48]: Found that this one is included on 1.5 branch, can you check?

File: UPDATING.md (hunk 1 above)
Comment (villebro) [reply UPDATING.md:46]: You're right, this one is included. I added it to the 1.5.0 changelog (will also add it to the 1.5 branch)

File: UPDATING.md (hunk 2 above)
Comment (villebro) [reply UPDATING.md:48]: This one is actually already included in the 1.4.1 release, so I'm removing it from the `Next` section.

**Review by dpgaspar — APPROVED** [review]

**Pull request merged**